integration_test/tests/odbc_2_tests.rs

```diff
@@ -0,0 +1,318 @@
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+            let mut table_view: Vec<WideChar> = cstr::to_widechar_vec("TABLE");
+            table_view.push(0);
+
+            // list tables with null pointers for the table strings
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLTablesW(
+                    stmt as HStmt,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    table_view.as_ptr(),
+                    table_view.len() as SmallInt - 1
+                ),
+                "{}",
+                get_sql_diagnostics(HandleType::Env, env_handle as Handle)
+            );
+
+            // assert all tables are returned from the previous SQLTables call
+            for _ in 0..14 {
+                assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as HStmt));
+            }
+
+            assert_eq!(SqlReturn::NO_DATA, SQLFetch(stmt as HStmt))
+        }
+    }
+
+    const EXPECTED_DATATYPES: [SqlDataType; 22] = [
+        SqlDataType::EXT_W_VARCHAR,
+        SqlDataType::EXT_BIT,
+        SqlDataType::EXT_BIG_INT,
+        SqlDataType::EXT_BINARY,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::UNKNOWN_TYPE,
+        SqlDataType::INTEGER,
+        SqlDataType::DOUBLE,
+        SqlDataType::EXT_TIMESTAMP,
+        SqlDataType::TIMESTAMP,
+    ];
+
+    /// call SQLGetTypeInfo to verify the correct types are returned. For all types,
+    /// we should get both date types back. For date, we should get the specific date type
+    /// we expect back.
+    #[test]
+    fn test_type_listing() {
+        let env_handle = setup();
+        let (conn_handle, _, _, _) = connect(env_handle);
+        let mut stmt: Handle = null_mut();
+
+        let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // check that when requesting all types, both odbc 2 and 3 timestamp types are returned
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::UNKNOWN_TYPE)
```

Comment:
Can you please add SQL_ALL_TYPES in definition or in common?
That both UNKNOWN_TYPE and SQL_ALL_TYPES identifiers are `0` is good and probably a little bit on purpose, but having the type called SQL_ALL_TYPES is much clearer.

Note : It is also defined in sql.h. Both are.